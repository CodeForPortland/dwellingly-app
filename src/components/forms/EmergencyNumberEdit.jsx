import React from 'react';
import PropTypes from 'prop-types';

const EmergencyNumberEdit = (props) => {
  const onSubmit = () => {
    props.onSavingNewEmergencyNumber();
    props.onCancellingEditEmergencyNumber();
  }
  return (
    <div id={props.id}>
      <input name="title" placeholder={props.title} onChange={props.onEditingEmergencyNumber} type="text" />
      <input name="number01" placeholder={props.number01} onChange={props.onEditingEmergencyNumber} type="text" />
      {props.number02 !== null && props.number02 !== ''
        && (
          <input name="number02" placeholder={props.number02} onChange={props.onEditingEmergencyNumber} type="text" />
        )
      }
      <button type="submit" className="btn" onClick={() => onSubmit()}>Save</button>
      <button type="button" className="btn btn--danger" onClick={props.onCancellingEditEmergencyNumber}>Cancel</button>
    </div>
  )
};

EmergencyNumberEdit.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  number01: PropTypes.string.isRequired,
  number02: PropTypes.string,
  onCancellingEditEmergencyNumber: PropTypes.func.isRequired,
  onEditingEmergencyNumber: PropTypes.func.isRequired,
  onSavingNewEmergencyNumber: PropTypes.func.isRequired
}

EmergencyNumberEdit.defaultProps = {
  number02: null
}

export default EmergencyNumberEdit
