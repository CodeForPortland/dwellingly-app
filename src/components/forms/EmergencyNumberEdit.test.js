import React from 'react';
import renderer from 'react-test-renderer';
import EmergencyNumberEdit from './EmergencyNumberEdit';

const EmergencyNumberEditTest = (props) =>
<EmergencyNumberEdit
{...props}
/>;


describe('EmergencyNumberEdit Component', () => {
  const props = {
    numberOne: '',
    numberTwo: '',
    numberThree: '',
  }
  it('correctly renders EmergencyNumberEdit component', () => {
    const EmergencyNumberEditComponent = renderer.create(<EmergencyNumberEdit {...props} />).toJSON();
    expect(EmergencyNumberEditComponent).toMatchSnapshot();
  });
  it('has a div with className emergencyNumberEditContainer', () => {
    const firstDiv = shallow(<EmergencyNumberEdit {...props} />).find('div').at(0);
    expect(firstDiv.hasClass('emergencyNumberEditContainer')).toEqual(true);
  });
  // it('user text is echoed onChange', () => {
  //   const wrapper = shallow(<EmergencyNumberEdit {...props} onEditingEmergencyText={() => {}} />)
  //   wrapper.find('input').at(0).simulate('change', {
  //     target: { value: '503-888-1234' }
  //   });
  //   expect(wrapper.find('input').at(0).props().value).toEqual('503-888-1234');
  // });
});
