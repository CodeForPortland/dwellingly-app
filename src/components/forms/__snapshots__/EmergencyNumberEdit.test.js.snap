// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmergencyNumberEdit Component correctly renders EmergencyNumberEdit component 1`] = `
<div
  className="emergencyNumberEditContainer"
>
  <input
    name="contact"
    type="text"
  />
  <input
    name="subtext"
    placeholder=""
    type="text"
  />
  <div
    className="emergencyNumberEditInputs"
  >
    <fieldset
      name="phoneNumberOne"
    >
      <input
        name="subtext"
        type="text"
      />
      <input
        name="number"
        type="text"
      />
      <input
        name="ext"
        type="text"
      />
    </fieldset>
  </div>
  <div
    className="btn--group"
  >
    <div
      className="btn--group-inner"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="submit"
      >
        Save
      </button>
      <button
        className="btn btn--disabled"
        type="button"
      >
        Delete
      </button>
    </div>
    <button
      className="btn btn--danger"
      type="button"
    >
      Cancel
    </button>
  </div>
</div>
`;
